llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp

```diff
@@ -7939,7 +7939,7 @@ ParseStatus AArch64AsmParser::tryParseSVEPattern(OperandVector &Operands) {
 
     auto *MCE = dyn_cast<MCConstantExpr>(ImmVal);
     if (!MCE)
-      return ParseStatus::Failure;
+      return TokError("expected immediate operand");
```

Comment:
since this function seems to parse a specific enum value, would not be better to emit a more specific error?

Reply:
The problem is that the instruction could have:
1- Hash followed by an immediate or
2- No hash, just an sve pattern
Is this fine:
"Unexpected non-immediate after the hash"
or  "Unexpected hash followed by a non-immediate"
?
Do you have another suggestion?

Reply:
Given the rational for the fix the chances are the `#` is erroneous rather than the user really wanting to use the `#imm` syntax so for consistency I suggest we emit the same `invalid operand for instruction` message we use for predicate pattern typos.  If at a later date we want to be more expressive then we can fix all instances at the same time.

Reply:
I was rather thinking about sve specific message for example: 
"operand must be an enumeration value describing number of active elements.”
the issue only occurs with the "all"
seems like both “ptrue p0.d, #1" and “ptrue p0.d, 1” are parsed as legal.

Reply:
An immediate operand is legal for the `ptrue` instruction, `#` indicates an immediate must follow:
https://developer.arm.com/documentation/ddi0602/2023-12/SVE-Instructions/PTRUE--predicate---Initialise-predicate-from-named-constraint-?lang=en#sa_pattern